feat(form-plugin): implement `propertyPath` parameter in the `UpdateFormValue` (#1215)
ngxs/store: docs/plugins/form.md

The form plugin comes with the following `actions` out of the box:

- `UpdateFormStatus({ status, path })` - Update the form status

> - `UpdateFormValue({ value, path, propertyPath? })` - Update the form value

Reviewer: Missed one tweak
```suggestion
- `UpdateFormValue({ value, path, propertyPath? })` - Update the form value (or optionally an inner property value)
```

- `UpdateFormDirty({ dirty, path })` - Update the form dirty status
- `SetFormDisabled(path)` - Set the form to disabled
- `SetFormEnabled(path)` - Set the form to enabled
- `SetFormDirty(path)` - Set the form to dirty (shortcut for `UpdateFormDirty`)
- `SetFormPristine(path)` - Set the form to pristine (shortcut for `UpdateFormDirty`)

> ### Updating Form Properties Separately

Reviewer: ```suggestion
### Updating Specific Form Properties
```

> The form plugin exposes `UpdateFormValue` action that gives the opportunity to update nested form properties by providing `propertyPath` parameter.

Reviewer: ```suggestion
The form plugin exposes the `UpdateFormValue` action that provides the ability to update nested form properties by supplying a `propertyPath` parameter.
```

The form plugin comes with the following `actions` out of the box:

- `UpdateFormStatus({ status, path })` - Update the form status

> - `UpdateFormValue({ value, path, propertyPath })` - Update the form value

Reviewer: ```suggestion
- `UpdateFormValue({ value, path, propertyPath? })` - Update the form value (or optionally an inner property value)
```

Reviewer: Didn't quite get all of this.

- `UpdateFormDirty({ dirty, path })` - Update the form dirty status
- `SetFormDisabled(path)` - Set the form to disabled
- `SetFormEnabled(path)` - Set the form to enabled
- `SetFormDirty(path)` - Set the form to dirty (shortcut for `UpdateFormDirty`)
- `SetFormPristine(path)` - Set the form to pristine (shortcut for `UpdateFormDirty`)

> ### Updating form properties separately

Reviewer: ```suggestion
### Updating Form Properties Separately
```

      novelName: string;
      authors: {
        name: string;
      }[];
    };
  };
}

@State<NovelsStateModel>({
  name: 'novels',
  defaults: {
    newNovelForm: {
      model: undefined
    }
  }
})
export class NovelsState {}
```

The state contains information about the new novel name and its authors. Let's create a component
that will render the reactive form with bounded `ngxsForm` directive:

```ts
@Component({
  selector: 'new-novel-form',
  template: `
    <form [formGroup]="newNovelForm" ngxsForm="novels.newNovelForm" (ngSubmit)="onSubmit()">
      <input type="text" formControlName="novelName" />

      <div
        formArrayName="authors"
        *ngFor="let author of newNovelForm.get('authors').controls; index as index"
      >
        <div [formGroupName]="index">
          <input formControlName="name" />
        </div>
      </div>

      <button type="submit">Create</button>
    </form>
  `
})
export class NewNovelComponent {
  newNovelForm = new FormGroup({
    novelName: new FormControl('Zenith'),
    authors: new FormArray([
      new FormGroup({
        name: new FormControl('Sasha Alsberg')
      })
    ])
  });

  onSubmit() {
    //
  }
}
```

> Let's look at the component above again. Assume we want to update the first control's `name`. The code would look as follows:

Reviewer: ```suggestion
Let's look at the component above again. Assume we want to update the name of the first author in our form, from anywhere in our application. The code would look as follows:
```
